src/main/java/org/alfresco/repo/management/JmxDumpUtil.java

```diff
@@ -164,9 +170,70 @@ private static void printMBeanInfo(MBeanServerConnection connection, ObjectName
                 attributes.put(OS_NAME, updateOSNameAttributeForLinux(osName));
             }
         }
+        if (objectName.getCanonicalName().equals("java.lang:type=Runtime"))
+        {
+            String[] commandInputs = (String[]) attributes.get(INPUT_ARGUMENTS);
+            if(commandInputs != null)
+            {
+                attributes.put(INPUT_ARGUMENTS, cleanPasswordsFromInputArguments(commandInputs));
+            }
+        }
         tabulate(JmxDumpUtil.NAME_HEADER, JmxDumpUtil.VALUE_HEADER, attributes, out, 0);
     }
 
+    /**
+     * If any of the string contain passwords, denote by "*password=" or "*pwd=" or "token="
+     * then they shall be replaces with JmxDumpUtil.PROTECTED_VALUE.
+     * @param commandInputs
+     * @return array of password redacted strings
+     */
+    public static String[] cleanPasswordsFromInputArguments(String[] commandInputs)
+    {
+        List<String> cleanInputs = new ArrayList<String>();
+        for (String input : commandInputs) 
+        {
+            input = cleanPasswordFromInputArgument(input);
+            cleanInputs.add(input);
+        }
+        
+        return cleanInputs.toArray(new String[commandInputs.length]);
+    }
+
+    /**
+     * Removes any characters following the words:
+     * password
+     * pwd
+     * token
+     * and replaces them with JmxDumpUtil.PROTECTED_VALUE
+     * 
+     * Example: 
+     * Input:   -Ddb.password=alfresco
+     * Output:  -Ddb.password=JmxDumpUtil.PROTECTED_VALUE
+     * 
+     * @param input
+     * @return password redacted string
+     */
+    public static String cleanPasswordFromInputArgument(String input)
+    {
+        // Regex to select the characters following the words password=, token= or pwd=
+        String pattern = "(?i)((?<=(password)=)|(?<=(token)=)|(?<=(pwd)=)).++";
+        String replacement = JmxDumpUtil.PROTECTED_VALUE;
+        String output = input.replaceAll(pattern, replacement);
+
+        if(output.equals(input)) //If nothing has been replaced it's possible password= could be end of string
+        {
+            // pattern to see if the string ends with "password=", "token=" or "pwd="
+            Pattern passwordAtEnd = Pattern.compile("((password=)|(token=)|(pwd=))$");

```

Comment:
The pattern compilation is very expensive, it would be better to do this outside of this method. 
Also, can we configure this list of patterns or prop names somewhere? Maybe there is already a similar list defined?

Reply:
I agree with @eknizat, the pattern compiling can be done once, doesn't have to be done for every call. As for having a separate list, we do have a protected attributes list, but that applies to the mbean we manage. In this case this filters only the input arguments part of the default JVM mbean.
I think this covers what we want for now.
@skopf what do think?

Reply:
I have removed the used of `Pattern.compile` and modified the implementation of this method. The regex used now has its own creation method `createPasswordFindRegexString(String[] argEndings)` that can dynamically create the required regex, based on the argument provided.